applications/HDF5Application/custom_python/add_custom_io_to_python.cpp

```diff
@@ -0,0 +1,90 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+//
+
+
+// System includes
+
+// External includes
+#include <boost/python.hpp>
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/io.h"
+
+// Application includes
+#include "custom_io/hdf5_file.h"
+#include "custom_io/hdf5_file_serial.h"
+#include "custom_io/hdf5_model_part_io.h"
+#include "custom_io/hdf5_nodal_solution_step_data_io.h"
+#ifdef KRATOS_USING_MPI
+#include "custom_io/hdf5_file_parallel.h"
+#include "custom_io/hdf5_partitioned_model_part_io.h"
+#endif
+
+namespace Kratos
+{
+
+namespace Python
+{
+
+void AddCustomIOToPython()
+{
+    using namespace boost::python;
+
+    class_<HDF5::File, HDF5::File::Pointer, boost::noncopyable >("HDF5File", init<Parameters&>())
+    .def("HasPath",&HDF5::File::HasPath)
+    .def("IsGroup",&HDF5::File::IsGroup)
+    .def("IsDataSet",&HDF5::File::IsDataSet)
+    .def("CreateGroup",&HDF5::File::CreateGroup)
+    .def("AddPath",&HDF5::File::AddPath)
+    .def("GetDataDimensions",&HDF5::File::GetDataDimensions)
+    .def("HasIntDataType",&HDF5::File::HasIntDataType)
+    .def("HasFloatDataType",&HDF5::File::HasFloatDataType)
+    .def("Flush",&HDF5::File::Flush)
+    .def("GetFileSize",&HDF5::File::GetFileSize)
+    .def("GetFileName",&HDF5::File::GetFileName)
+    ;
+
+    class_<HDF5::FileSerial, HDF5::FileSerial::Pointer, bases<HDF5::File>, boost::noncopyable>(
```

Comment:
I understand why you placed everything on a different namespace, but it generates some confusion with the python interface, since this one does not follow the same convention and just prepends `HDF5` to class names. What would you think of adding all these functions to an `hdf5` module in Python to replicate the same structure?

Reply:
Do you mean in addition to the KratosHDF5Application to have something like a KratosHDF5 module and access the classes like KratosHDF5.SerialFile()? I'm not sure how this can be done. I find the boost documentation on the python interface quite limited. Do you know an example or some reference that does something like what you have in mind? Alternatively, I could just remove the HDF5 prefix from the python class names and use the single module as a namespace. Would this be better?

Reply:
I see two possibilities here: either we rename the boost python module (which can be done but has some annoying details I don't remember right now) or we do something custom in the HDF5Application.py to place these functions in a different namespace. In any case, I can take care of that if we decide we want to do it like this.
In any case, I prefer to keep it as it is now rather than using only the module name (I think this is a bit error-prone, since we are not so consistent in importing applications with relative scopes in kratos).

Maybe this is not so important right now, it is just a discussion I wanted to have.

Reply:
ok, I think it's good to keep in mind, but I will not modify it here.